solutions/src/main/java/ru/sberbank/school/task08/state/MapState.java

```diff
@@ -0,0 +1,47 @@
+package ru.sberbank.school.task08.state;
+
+import java.util.List;
+
+public class MapState<T extends InstantiatableEntity> implements Savable<T> {
+    private final String name;
+    private final List<T> gameObjects;
+
+    private MapState() {
+        gameObjects = null;
+        name = null;
+    }
+
+    public MapState(String name, List<T> gameObjects) {
+        this.name = name;
+        this.gameObjects = gameObjects;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<T> getGameObjects() {
+        return this.gameObjects;
+    }
+
+    public boolean equals(final Object o) {
+        if (o == this) return true;
+        if (!(o instanceof MapState)) return false;
+        final MapState other = (MapState) o;
+        final Object this$gameObjects = this.getGameObjects();
+        final Object other$gameObjects = other.getGameObjects();
+        return this$gameObjects == null ? other$gameObjects == null : this$gameObjects.equals(other$gameObjects);
+    }
+
+    public int hashCode() {
+        final int PRIME = 59;
```

Comment:
Abbreviation in name &#39;PRIME&#39; must contain no more than &#39;2&#39; consecutive capital letters.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-844129236>*